apps/playlist/pkg/watchers/reconciler_playlist.go

```diff
@@ -0,0 +1,62 @@
+package watchers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"k8s.io/klog/v2"
+
+	"github.com/grafana/grafana-app-sdk/operator"
+	"github.com/grafana/grafana-app-sdk/resource"
+	"github.com/grafana/grafana-app-sdk/simple"
+
+	playlist "github.com/grafana/grafana/apps/playlist/pkg/apis/playlist/v0alpha1"
+)
+
+func NewPlaylistReconciler() operator.Reconciler {
+	inner := simple.Reconciler{}
+
+	inner.ReconcileFunc = func(ctx context.Context, request operator.ReconcileRequest) (operator.ReconcileResult, error) {
+		switch request.Action {
+		case operator.ReconcileActionCreated:
+			pObj, err := assertType(request.Object)
+			if err == nil {
+				klog.InfoS("Added resource", "name", pObj.GetStaticMetadata().Identifier().Name)
+			}
+			return operator.ReconcileResult{}, err
+		case operator.ReconcileActionUpdated:
+			pObj, err := assertType(request.Object)
+			if err == nil {
+				klog.InfoS("Updated resource", "name", pObj.GetStaticMetadata().Identifier().Name)
+			}
+			return operator.ReconcileResult{}, err
+		case operator.ReconcileActionDeleted:
+			pObj, err := assertType(request.Object)
+			if err == nil {
+				klog.InfoS("Deleted resource", "name", pObj.GetStaticMetadata().Identifier().Name)
+			}
+			return operator.ReconcileResult{}, err
+		case operator.ReconcileActionResynced:
+			pObj, err := assertType(request.Object)
+			if err == nil {
+				klog.InfoS("Possibly updated resource", "name", pObj.GetStaticMetadata().Identifier().Name)
+			}
+			return operator.ReconcileResult{}, err
+		case operator.ReconcileActionUnknown:
+			return operator.ReconcileResult{}, errors.New("Invalid reconcile action unknown")
```

Comment:
`simple.App` uses an `operator.InformerController`, [which will retry errors returned from the reconciler according to its RetryPolicy](https://github.com/grafana/grafana-app-sdk/blob/main/operator/informer_controller.go#L562-L570). By default this policy is an [ExponentialBackoffRetryPolicy which will do 5 total retries](https://github.com/grafana/grafana-app-sdk/blob/main/operator/informer_controller.go#L35). To avoid retrying errors like this which will not be resolved on a retry, we may want to either set up a custom RetryPolicy that doesn't retry these errors in [AppConfig.InformerConfig](https://github.com/grafana/grafana-app-sdk/blob/main/simple/app.go#L114) (perhaps use a type which implements `error` and some `Retryable() bool` function which can be checked in the RetryPolicy), or log the error and return `nil` so the ReconcileRequest won't be retried later.

Reply:
Thank you. Based on that, I will go with:

> log the error and return nil so the ReconcileRequest won't be retried later.